parquet-scala/src/main/scala/parquet/filter2/dsl/Dsl.scala

```diff
@@ -26,10 +27,10 @@ import parquet.io.api.Binary
  */
 object Dsl {
 
-  private[Dsl] trait Column[T <: Comparable[T], C <: Operators.Column[T]] {
+  private[Dsl] trait Column[T <: Comparable[T], C <: Operators.Column[T], S <: java.io.Serializable] {
     val javaColumn: C
 
-    def filterBy[U <: UserDefinedPredicate[T]](clazz: Class[U]) = FilterApi.userDefined(javaColumn, clazz)
+    def filterBy[U <: UserDefinedPredicate[T, S]](clazz: Class[U], o: S) = FilterApi.userDefined(javaColumn, clazz, o)
```

Comment:
S could be defined as a parameter of this method rather then in Column. That would avoid having to add Serializable everywhere bellow